docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,392 @@
+
+1. The ShardedTensorFactories are applied
+2. LocalNonPersistentObject are extracted from the sharded state dict and ignored
+3. ShardedBase objects are extracted
+4. All other objects are treated as "common" and saved according to a sharded strategy (see `Save and load strategies`_)
+5. All ShardedObjects are extracted from point (3) objects and saved with a common strategy (see `Save and load strategies`_)
+6. All ShardedTensors are saved.
+7. `metadata.json` file with backend and version metadata is saved to the checkpoint directory.
+
+dist_checkpointing.load
+-----------------------
+The main entrypoint for checkpoint loading.
+Requires providing a sharded state dict (in order to implicitly define mappings between local tensors and checkpoint tensors) and loading strategies.
+In practice, the same sharded state dict can be usually used for both saving and loading (the sharded state dict for loading will just contain tensors with uninitialized data).
+
+The sharded state dict provided as an input is processed in the following way:
+
+1. "common" state dict is loaded from the checkpoint. This forms the base of the resulting state dict
+2. The ShardedTensorFactories from the input sharded state dict are applied
+3. LocalNonPersistentObject are extracted from the input sharded state dict, unwrapped and added to the resulting state dict
+4. ShardedObjects are extracted and loaded from the checkpoint into the resulting state dict
+5. ShardedTensors are extracted and loaded from the checkpoint into the resulting state dict
+6. Factory merges are applied (see `Optimizers`_ for explanation)
+
+This results in a *regular* state dict with plain tensors that can be further processed by the application (which usually means running `model.load_state_dict(state_dict)`.
+
+
+dist_checkpointing.load_common_state_dict
+-----------------------------------------
+An entrypoint allowing loading ony the "common" part of the checkpoints.
+Most of the checkpoint config and metadata can be loaded with this method, which allows skipping data loading in order to take decisions regarding checkpoint config, version, etc.
+
+dist_checkpointing.load_tensors_metadata
+----------------------------------------
+This entrypoint allows to read all ShardedTensors metadata from the checkpoint without loading any data.
+The result is a sharded state dict with trivial sharding (every tensor is sharded into one big shard).
+
+dist_checkpointing.load_plain_tensors
+-------------------------------------
+This entrypoint allows to read sharded tensors stored in the checkpoint without any sharding (as plain tensors).
+This function is simply a composition of `load_tensors_metadata` and `save`.
+
+Save and load strategies
+========================
+There are multiple ways to save a sharded state dict into a serialized checkpoint. They can be provided by the user as saving and loading strategies.
+
+There are four types of strategies:
```

Comment:
There are several methods for saving a sharded state dictionary into a serialized checkpoint. These methods can be specified by the user as saving and loading strategies. There are four types of strategies available: